### PR TITLE
finished k-nth code challenge

### PR DESCRIPTION
**Pull request by wosunkwo**

### CODE REVIEW

**Comment by wosunkwo** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -135,13 +135,51 @@ public int insertAfter(int value, int newVal){
...
                 }
             }
         }
         System.out.println("The linked list does not contain "+ value);
+        return;
+    }
+
+    public int knthFromEnd(int value){
+        if(this.head == null){
+            throw new NullPointerException("The Linked list is empty");
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java:150]: You don't always want to throw errors! It's perfectly reasonable to pass in an empty LinkedList into this function or pass a value that is larger than the length of the LinkedList. Instead of throwing an error (which will cause the code to stop) it would be better to handle the issue so your code will continue to run

File: code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java [hunk 2]
```diff
@@ -135,13 +135,51 @@ public int insertAfter(int value, int newVal){
...
+            while(current != null){
+                if(kIndex == listCounter){
+                    return current.value;
+                }else{
+                    current = current.next;
+                    listCounter++;
+                }
+            }
+        }
         return -1;
```
Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java]: return `-1` here is a bad idea. The user won't know if `-1` is the value k-the-from-the-end or if `-1` means that the value wasn't found

Comment (evanslaton) [inline code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java:172]: I misread the code earlier, while your code should never actually return `-1` I'm not sure it is wise to return `-1`
